File: Oficinas/29-07-22 Introdução à Programação Competitiva/Códigos/Ida a Feira/feira.py
t = int(input())

for k in range(t):  #para cada caso de teste k = 0, 1, ..., t-1
    precos = {} #instaciando o map (chamado de dict, no Python)
    total = 0

    n = int(input())
    #Lendo cada um dos itens e seus valores unitários
    for i in range(n):
        item, valor = input().split()
        precos[item] = float(valor) #Armazenando no dict
    
    m = int(input())
    #Lendo a lista de compras
    for i in range(m):
        item, qtde = input().split()
        total = total + precos[item] * int(qtde)
        
    print("R$ {:.2f}".format(total))

File: Oficinas/29-07-22 Introdução à Programação Competitiva/Códigos/Ida a Feira/feira.cpp
#include<bits/stdc++.h>

using namespace std;

int main(){
    int t, i;

    cin >> t;

    while(t--)  //Para cada caso de teste: Enquanto t > 0, decrementa o t e executa o laço. Quando t == 0, encerra
    { 
        int n, m, qtde;
        map<string, double> precos; //map para armazenar os preços de cada item, em que o nome do item é a chave
        string item;
        double valor, total = 0;

        cin >> n;
        //Lendo cada um dos itens e seus valores unitários
        for(i = 0; i < n; i++){
            cin >> item >> valor;
            precos[item] = valor;   //Armazenando no map
        }

        //Lendo a lista de compras
        cin >> m;
        for(i = 0; i < m; i++){
            cin >> item >> qtde;
            total += precos[item] * qtde;
        }

        printf("R$ %.2lf\n", total);
    }
}